src/statemachine.rs

```diff
@@ -412,38 +415,37 @@ impl StateMachineCtap2 {
         }
     }
 
-    fn determine_pin_auth<T: PinAuthCommand, U>(
+    fn determine_pin_auth<T: PinUvAuthCommand + Request<V>, U, V>(
         cmd: &mut T,
         dev: &mut Device,
+        skip_uv: bool,
         status: &Sender<StatusUpdate>,
         callback: &StateCallback<crate::Result<U>>,
-    ) -> Result<(), ()> {
-        loop {
-            match cmd.determine_pin_auth(dev) {
-                Ok(_) => {
-                    break;
+    ) -> Result<PinUvAuthResult, PinUvAuthError> {
```

Comment:
nit: I would pass `alive` into this and handle the retry logic here. Then we only return an `Err` for unrecoverable errors.